monkey/tests/unit_tests/monkey_island/cc/agent_event_handlers/test_scan_event_handler.py

```diff
@@ -125,144 +161,176 @@ def __call__(self, ip: IPv4Address):
             return machines
 
 
-def test_handle_ping_scan_event__target_machine_not_exists(
-    handler: handle_ping_scan_event,
-    machine_repository: IMachineRepository,
+@pytest.mark.parametrize(
+    "event,handler",
+    [(PING_SCAN_EVENT, "handle_ping_scan_event"), (TCP_SCAN_EVENT, "handle_tcp_scan_event")],
+)
+def test_scan_event_handler__target_machine_not_exists(
+    event, handler, machine_repository: IMachineRepository, request
 ):
     machine_repository.get_machine_by_id = MagicMock(side_effect=machine_from_id)
     machine_repository.get_machines_by_ip = MagicMock(side_effect=UnknownRecordError)
 
-    handler(EVENT)
-
-    expected_machine = Machine(id=SEED_ID, network_interfaces=[IPv4Interface(EVENT.target)])
-    expected_machine.operating_system = EVENT.os
-    machine_repository.upsert_machine.assert_called_with(expected_machine)
-
-
-def test_handle_ping_scan_event__target_machine_already_exists(
-    handler: handle_ping_scan_event,
-    machine_repository: IMachineRepository,
-):
-    machine_repository.get_machine_by_id = MagicMock(side_effect=machine_from_id)
-    machine_repository.get_machines_by_ip = MagicMock(side_effect=machines_from_ip)
-
-    handler(EVENT)
+    handler = request.getfixturevalue(handler)
```

Comment:
Interesting -- so this is what we need to do if we want to pass fixtures to `pytest.mark.parametrize`

Reply:
I think we can do this with `indirect=True` and clean it up. I'll experiment.